tripy/tripy/config.py

```diff
@@ -27,22 +27,28 @@
 
 export.public_api(autodoc_options=[":no-members:", ":no-special-members:"])(sys.modules[__name__])
 
-# MLIR Debug options
+# MLIR-TRT Debug options
 enable_mlir_debug = os.environ.get("TRIPY_MLIR_DEBUG_ENABLED", "0") == "1"
 mlir_debug_types = os.environ.get("TRIPY_MLIR_DEBUG_TYPES", "-mlir-print-ir-after-all,-translate-to-tensorrt").split(
     ","
 )
 mlir_debug_tree_path = os.environ.get("TRIPY_MLIR_DEBUG_PATH", os.path.join("/", "tripy", "mlir-dumps"))
 
-# Tensorrt debug options
+# TensorRT debug options
 enable_tensorrt_debug = os.environ.get("TRIPY_TRT_DEBUG_ENABLED", "0") == "1"
 tensorrt_debug_path = os.environ.get("TRIPY_TRT_DEBUG_PATH", os.path.join("/", "tripy", "tensorrt-dumps"))
 
-# Variables that are exposed to the user are kept lowercase.
 timing_cache_file_path: str = export.public_api(
     document_under="config.rst",
     autodoc_options=[":no-value:"],
     module=sys.modules[__name__],
     symbol="timing_cache_file_path",
 )(os.path.join(tempfile.gettempdir(), "tripy-cache"))
-"""Path to a timing cache file that can be used to speed up compilation time"""
+"""Path to a timing cache file that can be used to speed up compilation time."""
+
+enable_dtype_checking: bool = export.public_api(
```

Comment:
It does add overhead in eager mode, especially since it's applied to every single API, but more importantly, we need a mechanism to disable it for the dtype constraint tests. Otherwise, the negative testing becomes useless because data types we claim not to support will fail in the decorator itself.